crates/solidity/outputs/cargo/crate/src/backend/ast/generated/mutator.rs

```diff
@@ -0,0 +1,2963 @@
+        source: &YulVariableDeclarationStatement,
+    ) -> YulVariableDeclarationStatement {
+        let variables = self.mutate_yul_variable_names(&source.variables);
+        let value = source
+            .value
+            .as_ref()
+            .map(|value| self.mutate_yul_variable_declaration_value(value));
+
+        Rc::new(YulVariableDeclarationStatementStruct {
+            node_id: source.node_id,
+            variables,
+            value,
+        })
+    }
+
+    fn mutate_yul_variable_declaration_value(
+        &mut self,
+        source: &YulVariableDeclarationValue,
+    ) -> YulVariableDeclarationValue {
+        let assignment = self.mutate_yul_assignment_operator(&source.assignment);
+        let expression = self.mutate_yul_expression(&source.expression);
+
+        Rc::new(YulVariableDeclarationValueStruct {
+            node_id: source.node_id,
+            assignment,
+            expression,
+        })
+    }
+
+    fn mutate_yul_variable_assignment_statement(
+        &mut self,
+        source: &YulVariableAssignmentStatement,
+    ) -> YulVariableAssignmentStatement {
+        let variables = self.mutate_yul_paths(&source.variables);
+        let assignment = self.mutate_yul_assignment_operator(&source.assignment);
+        let expression = self.mutate_yul_expression(&source.expression);
+
+        Rc::new(YulVariableAssignmentStatementStruct {
+            node_id: source.node_id,
+            variables,
+            assignment,
+            expression,
+        })
+    }
+
+    fn mutate_yul_if_statement(&mut self, source: &YulIfStatement) -> YulIfStatement {
+        let condition = self.mutate_yul_expression(&source.condition);
+        let body = self.mutate_yul_block(&source.body);
+
+        Rc::new(YulIfStatementStruct {
+            node_id: source.node_id,
+            condition,
+            body,
+        })
+    }
+
+    fn mutate_yul_for_statement(&mut self, source: &YulForStatement) -> YulForStatement {
+        let initialization = self.mutate_yul_block(&source.initialization);
+        let condition = self.mutate_yul_expression(&source.condition);
+        let iterator = self.mutate_yul_block(&source.iterator);
+        let body = self.mutate_yul_block(&source.body);
+
+        Rc::new(YulForStatementStruct {
+            node_id: source.node_id,
+            initialization,
+            condition,
+            iterator,
+            body,
+        })
+    }
+
+    fn mutate_yul_switch_statement(&mut self, source: &YulSwitchStatement) -> YulSwitchStatement {
+        let expression = self.mutate_yul_expression(&source.expression);
+        let cases = self.mutate_yul_switch_cases(&source.cases);
+
+        Rc::new(YulSwitchStatementStruct {
+            node_id: source.node_id,
+            expression,
+            cases,
+        })
+    }
+
+    fn mutate_yul_default_case(&mut self, source: &YulDefaultCase) -> YulDefaultCase {
+        let body = self.mutate_yul_block(&source.body);
```

Comment:
discussed offline:
I suggest checking the `Rc<>` raw value of the returned children. If the implementation returns the same children unmutated, it means we can also return the parent without cloning values/creating a new parent.